### DISCUSSION TITLE
argo-workflows/managed-namespace/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9497] [General]
# argo-workflows/managed-namespace/

https://argoproj.github.io/argo-workflows/managed-namespace/

**Comment by mzk008** [comment]
where to add this arguments?
And another question, how to make argo workflow create resource in another namespace?

**Reply by nicolas-vivot** [reply]
If you are using Argo Events to trigger your workflows based on different input sources, you can deploy your sensors in the target namespaces and workflow triggered by such sensor would get executed into the sensor namespace. That is one way to do it when using Argo Events. I believe you can do something similar with the web-hook system.

**Reply by nicolas-vivot** [reply]
You should also be able to do it by using the Workflow of Worflows pattern, adding a workflow parameters on the parent workflow for the target namespace. The parent workflow would then create the real workflow into the target namespace and monitor it. You would end with two workflow in the end, one being executed in a default namespace, the other in the target one. Not really a great design but that would definitely work.

[a comment by an automated account is not shown]

**Comment by nicolas-vivot** [comment]
Documentation is not clear about the consequences of a per namespace installation. For example, it requires both the server and the controller to be installed in each namespace. Does this means that there will be as much UI as installation / server running per namespace? Are there any potential impact for cluster-wide workflow templates? (or anything else)

**Comment by mozarik** [comment]
Documentation about multiple namespace and managed namespace is not clear and i cannot grasp any context of what the functionality of this configuration

**Reply by gtato** [reply]
I guess there are cases where you have multiple installations of argo workflows in your cluster, from different tenants. In that case you wouldn't want one argo workflow to mess with the workflows of another tenant.
